### PR TITLE
Update the docker to build the repo compiler with the default triple "x86_64-pc-linux-musl-repo".

### PR DESCRIPTION
**Pull request by MaggieYingYi**
This branch is dependent on the https://github.com/SNSystems/llvm-project-prepo/pull/170.

### CODE REVIEW

**Review by paulhuggett — COMMENTED** [review]
A few minor comments. It’s worth correcting the PR title (it’s -musl-repo not -gnu-repo).

File: llvm/utils/repo/docker/llvm-prepo/Dockerfile [hunk 1]
```diff
@@ -128,9 +128,26 @@ RUN mkdir -p $musl_install_dir/include/linux
...
+
+RUN mkdir ${install_dir}/lib/linux/
+RUN repo-create-ticket --output=${install_dir}/lib/linux/clang_rt.crtbegin-x86_64.o --repo=${stdlib_repo} d3540c32449fa62681dc5c22fbc2a353
+RUN repo-create-ticket --output=${install_dir}/lib/linux/clang_rt.crtend-x86_64.o --repo=${stdlib_repo} d3540c32449fa62681dc5c22fbc2a353
+
+RUN repo-create-ticket --output=${install_dir}/lib/linux/clang_rt.o --repo=${stdlib_repo} d3540c32449fa62681dc5c22fbc2a353
+RUN ar cr ${install_dir}/lib/linux/libclang_rt.builtins-x86_64.a ${install_dir}/lib/linux/clang_rt.o
+
+RUN repo-create-ticket --output=${install_dir}/lib/libc++.o --repo=${stdlib_repo} d3540c32449fa62681dc5c22fbc2a353
+RUN ar cr ${install_dir}/lib/libc++.a ${install_dir}/lib/libc++.o
```
[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/utils/repo/docker/llvm-prepo/Dockerfile:144]: Fixed in the commit https://github.com/SNSystems/llvm-project-prepo/pull/171/commits/c4f69d2fc2eaad07293e2086c7d864efd0e97f85.

File: llvm/utils/repo/docker/llvm-prepo/Dockerfile [hunk 2]
```diff
@@ -128,9 +128,26 @@ RUN mkdir -p $musl_install_dir/include/linux
 COPY llvm-prepo/futex.h $musl_install_dir/include/linux/futex.h
 RUN cp /usr/include/linux/version.h $musl_install_dir/include/linux/version.h
 
-#############
-# libunwind #
-#############
+########################
+# genearte dummy files #
```
[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/utils/repo/docker/llvm-prepo/Dockerfile]: Fixed in the commit https://github.com/SNSystems/llvm-project-prepo/pull/171/commits/c4f69d2fc2eaad07293e2086c7d864efd0e97f85.

File: llvm/utils/repo/docker/llvm-prepo/Dockerfile [hunk 3]
```diff
@@ -141,16 +158,37 @@ ARG cmake_common="\
...
+  -D CMAKE_ASM_COMPILER_TARGET=x86_64-pc-linux-gnu-elf     \
+  -D CMAKE_C_FLAGS="--sysroot=$install_dir"                \
+  -D CMAKE_CXX_FLAGS="--sysroot=$install_dir"              \
   -D CMAKE_INSTALL_PREFIX=$install_dir                     \
-  -D CMAKE_TOOLCHAIN_FILE=$workspace_dir/src/llvm/utils/repo/full_repo.cmake \
   -D LLVM_CONFIG_PATH=$workspace_dir/build/bin/llvm-config \
   -D LLVM_ENABLE_LIBCXX=Yes                                \
   -D LLVM_PATH=$workspace_dir/src/llvm                     \
   -D musl=Yes                                              \
   -D musl_install=$musl_install_dir                        \
```
[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/utils/repo/docker/llvm-prepo/Dockerfile]: Removed in the commit https://github.com/SNSystems/llvm-project-prepo/pull/171/commits/c4f69d2fc2eaad07293e2086c7d864efd0e97f85.

File: llvm/utils/repo/docker/llvm-prepo/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulhuggett) [reply llvm/utils/repo/docker/llvm-prepo/Dockerfile:144]: You don't need either the `repo-create-ticket` or the `ar cr` if you use `ar d`.

[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/utils/repo/docker/llvm-prepo/Dockerfile]: Thanks, fixed in https://github.com/SNSystems/llvm-project-prepo/pull/171/commits/c4c21bab7ce7cf8d1fbb84712911cf0718c0b899.

**Review by paulhuggett — APPROVED** [review]

**Pull request merged**